plugins/verifier/licensechecker/licensechecker.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/deislabs/ratify/plugins/verifier/licensechecker/utils"
+
+	"github.com/deislabs/ratify/pkg/common"
+	"github.com/deislabs/ratify/pkg/ocispecs"
+	"github.com/deislabs/ratify/pkg/referrerstore"
+	_ "github.com/deislabs/ratify/pkg/referrerstore/oras"
+	"github.com/deislabs/ratify/pkg/verifier"
+	"github.com/deislabs/ratify/pkg/verifier/plugin/skel"
+)
+
+type PluginConfig struct {
+	Name            string   `json:"name"`
+	AllowedLicenses []string `json:"allowedLicenses"`
+}
+
+type PluginInputConfig struct {
+	Config PluginConfig `json:"config"`
+}
+
+func main() {
+	skel.PluginMain("licensechecker", "1.0.0", VerifyReference, []string{"1.0.0"})
+}
+
+func parseInput(stdin []byte) (*PluginConfig, error) {
+	conf := PluginInputConfig{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse stdin for input: %v", err)
+	}
+
+	return &conf.Config, nil
+}
+
+func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, descriptor ocispecs.ReferenceDescriptor, store referrerstore.ReferrerStore) (*verifier.VerifierResult, error) {
+	input, err := parseInput(args.StdinData)
+	if err != nil {
+		return nil, err
+	}
+
+	allowedLicenses := utils.LoadAllowedLicenses(input.AllowedLicenses)
+
+	ctx := context.Background()
+	referenceManifest, err := store.GetReferenceManifest(ctx, subjectReference, descriptor)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, blobDesc := range referenceManifest.Blobs {
+		refBlob, err := store.GetBlobContent(ctx, subjectReference, blobDesc.Digest)
+		if err != nil {
+			return nil, err
+		}
+
+		spdxDoc, err := utils.BlobToSPDX(refBlob)
```

Comment:
would this be extended into formats other than spdx? if so, should this be generic?

Reply:
Are you referring to other formats of storing SPDX metadata (tag-value, spdx-json, cyclonedx, etc.), or other non-SPDX SBoM formats?

For SPDX metadata in formats other than tag-value I can update the implementation of `BlobToSPDX` to consider the `ArtifactType` from the descriptor. The `spdx-tools` library has support for tag-value, json, and rdf out of the box.